khelo/lib/ui/flow/tournament/detail/tabs/tournament_detail_stats_tab.dart

```diff
@@ -195,21 +195,58 @@ class TournamentDetailStatsTab extends ConsumerWidget {
 
   void showFilterOptionSelectionSheet(
     BuildContext context, {
-    required Function(KeyStatTag) onTap,
-    KeyStatTag? keyStatFilter,
+    required Function(KeyStatFilterTag) onTap,
+    KeyStatFilterTag? selectedTag,
   }) async {
-    return await showActionBottomSheet(
-        context: context,
-        items: KeyStatTag.values
-            .map((option) => BottomSheetAction(
-                  title: option.getString(context),
-                  enabled: keyStatFilter != option,
-                  showCheck: keyStatFilter == option,
+    return await showModalBottomSheet(
+      context: context,
+      showDragHandle: true,
+      enableDrag: true,
+      isScrollControlled: true,
+      useRootNavigator: true,
+      backgroundColor: context.colorScheme.surface,
+      builder: (context) {
+        return Container(
+          width: double.infinity,
+          padding: EdgeInsets.all(16)
+              .copyWith(bottom: context.mediaQueryPadding.bottom + 24),
+          child: Wrap(
+            alignment: WrapAlignment.start,
+            spacing: 8,
+            runSpacing: 8,
+            children: KeyStatFilterTag.values.map(
+              (element) {
+                final isSelected = selectedTag == element;
+                return OnTapScale(
                   onTap: () {
                     context.pop();
-                    onTap(option);
+                    onTap(element);
                   },
-                ))
-            .toList());
+                  child: Chip(
```

Comment:
let's see if it's repeating or not and if yes then let's create common widget for it (is not available already).